### ISSUE TITLE
Significant slowdown in Keras model.fit() on simple problem when using validation data

### ISSUE THREAD

**Issue by aaustin141** [keras-team/tf-keras#336]
**System information**.
- Have I written custom code:  No
- OS Platform and Distribution:  Ubuntu 18.04
- TensorFlow installed from:  Binary
- TensorFlow version:  2.11.0
- Python version:  3.10.8
- Bazel version:  N/A
- GPU model and memory:  NVIDIA Quadro GV100 (32 GB)
- Exact command to reproduce:  See below.

**Describe the problem**.

Per instructions I received from @mohantym, this is a cross-post to this
repository of the issue tensorflow/tensorflow#58828, which I opened yesterday.

The issue documents a significant slowdown in training speed when using Keras
model.fit() with the validation_data parameter set.  On my hardware, I see a
slowdown of approximately 10x.  On Colab, @mohantym sees a slowdown of only
3.5x, but this still seems excessive to me given that the validation data is
only 1/8 the size of the training data.  I would not expect it to take 3.5x
time to process 12.5% more data per epoch.

(In fact, I see the same slowdown even if I reduce the validation set size to a
single example, so it's a 3.5x to 10x speed penalty for an 0.05% increase in
the amount of data to be processed.)

For further details, including code that can be used to reproduce the problem,
please see the aforementioned issue.

**Describe the current behavior**.

A 3.5x-10x slowdown during training using model.fit() with validation_data set.

**Describe the expected behavior**.

Some slowdown (validation isn't free)---but a whole lot less than 3.5x-10x.

**Standalone code to reproduce the issue**.

See the aforementioned issue.

**Source code / logs**.

See the aforementioned issue.
